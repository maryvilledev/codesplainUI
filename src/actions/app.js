import axios from 'axios';

export const EDIT_ANNOTATION = 'EDIT_ANNOTATION';
export const PARSE_SNIPPET = 'PARSE_SNIPPET';
export const RESTORE_STATE = 'RESTORE_STATE';
export const SAVE_ANNOTATION = 'SAVE_ANNOTATION';
export const SET_AST = 'SET_AST';
export const SET_RULE_FILTERS = 'SET_RULE_FILTERS';
export const SET_SNIPPET_CONTENTS = 'SET_SNIPPET_CONTENTS';
export const SET_SNIPPET_LANGUAGE = 'SET_SNIPPET_LANGUAGE';
export const SET_SNIPPET_TITLE = 'SET_SNIPPET_TITLE';
export const TOGGLE_EDITING_STATE = 'TOGGLE_EDITING_STATE';
export const SAVE_STATE = 'SAVE_STATE';
export const SAVE_STATE_STARTED = 'SAVE_STATE_STARTED';
export const SAVE_STATE_SUCCEEDED = 'SAVE_STATE_SUCCEEDED';
export const SAVE_STATE_FAILED = 'SAVE_STATE_FAILED';


export const setSnippetContents = (snippet) => ({
  type: SET_SNIPPET_CONTENTS,
  payload: snippet,
});

export const setRuleFilters = (filters) => ({
  type: SET_RULE_FILTERS,
  payload: filters,
});

export const setAST = (AST) => ({
  type: SET_AST,
  payload: AST,
});

export const toggleEditState = () => ({
  type: TOGGLE_EDITING_STATE,
});

export const saveAnnotation = (annotationData) => ({
  type: SAVE_ANNOTATION,
  payload: annotationData,
});

export const restoreState = (savedState) => ({
  type: RESTORE_STATE,
  payload: savedState,
});

export const setSnippetTitle = (snippetTitle) => ({
  type: SET_SNIPPET_TITLE,
  payload: snippetTitle
})

export const setSnippetLanguage = (snippetLanguage) => ({
  type: SET_SNIPPET_LANGUAGE,
  payload: snippetLanguage
});

export const parseSnippet = (snippet) => ({
  type: PARSE_SNIPPET,
  meta: {
    WebWorker: true,
  },
  payload: {
    snippet,
  },
});

export const saveStateStarted = () => ({
  type: SAVE_STATE_STARTED,
});

export const saveStateSucceeded = () => ({
  type: SAVE_STATE_SUCCEEDED,
});

export const saveStateFailed = () => ({
  type: SAVE_STATE_FAILED,
});

export const saveState = (id) => {
  return (dispatch, getState) => {
    // If the snippet has not been saved, then saving an annotation shouldn't
    // trigger a POST request to save the snippet (and its annotations)
    if (id === undefined) {
      // Return a Promise object to make this action "thenable"
      return Promise.resolve();
    }
    dispatch(saveStateStarted());
    // Get the app state and stringify it
    const stateString = JSON.stringify(getState().app);

    // TODO: Dispatch action to show the snackbar with a relevant message based on the results
    // Return the results of the POST request
    return axios.post(`api/snippets/${id}`, { json: stateString, })
      .then((res) => {
        dispatch(saveStateSucceeded());
      }, (err) => {
        dispatch(saveStateFailed());
      });
  };
};
