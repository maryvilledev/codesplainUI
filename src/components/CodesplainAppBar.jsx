import React from 'react';
import AppBar from 'material-ui/AppBar';
import { browserHistory } from 'react-router';
import cookie from 'react-cookie';

import LoginButton from './buttons/LoginButton'
import AppMenu from './AppMenu';

const styles = {
  title: {
    cursor: 'pointer',
  },
}

/*
<CodesplainAppBar /> renders as the banner located at the top of the screen. It 
contains the text "Codesplain" on the far left, and either <AppMenu /> or
<LoginButton /> components on its right, depending on whether or not the user
has signed in with GitHub.
*/
class CodesplainAppBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoggedIn: cookie.load('token') !== undefined,
    }
    this.handleSignOut = this.handleSignOut.bind(this);
  }

  handleSignOut() {
    cookie.remove('token');
    this.setState({ isLoggedIn: false });
  }

  render() {
    const rightElement = this.state.isLoggedIn ?
                            <AppMenu onSignOut={this.handleSignOut} /> :
                            <LoginButton />
    return (
      <AppBar
        showMenuIconButton={false}
        title="Codesplain"
        style={styles.title}
        onTitleTouchTap={() => {
          browserHistory.push('/');
          location.reload();
        }}
        iconElementRight={rightElement}
      />
    );
  }

}

export default CodesplainAppBar;
