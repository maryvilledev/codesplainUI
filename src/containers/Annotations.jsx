import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { CardText } from 'material-ui/Card';

import {
  closeAnnotationPanel,
} from '../actions/annotation';

import {
  saveAnnotation,
  saveState,
} from '../actions/app';

import AnnotationPanel from '../components/AnnotationPanel';

export class Annotations extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      displayStatus: 'none',
    };
    this.handleCloseAnnotation = this.handleCloseAnnotation.bind(this);
    this.handleSaveAnnotation = this.handleSaveAnnotation.bind(this);
  }

  handleCloseAnnotation() {
    const { dispatch } = this.props;
    dispatch(closeAnnotationPanel());
  }

  handleSaveAnnotation(annotation) {
    const { dispatch, id, snippetInformation } = this.props;
    const annotationData = {
      annotation,
      ...snippetInformation,
    };
    dispatch(saveAnnotation(annotationData));
    dispatch(saveState(id));
  }

  render() {
    const {
      annotation,
      isDisplayingAnnotation,
      readOnly,
      snippetInformation,
    } = this.props;

    if (!isDisplayingAnnotation) {
      const prompt = readOnly ?
        'Click on a line number to add an annotation or display one' :
        'Lock this snippet to add annotations';
      return (
        <CardText>{prompt}</CardText>
      );
    }
    return (
      <AnnotationPanel
        annotation={annotation}
        snippetInformation={snippetInformation}
        saveAnnotation={this.handleSaveAnnotation}
        closeAnnotation={this.handleCloseAnnotation}
      />
    );
  }
}

Annotations.propTypes = {
  annotation: PropTypes.string.isRequired,
  isDisplayingAnnotation: PropTypes.bool.isRequired,
  readOnly: PropTypes.bool.isRequired,
  snippetInformation: PropTypes.shape({
    lineNumber: PropTypes.number,
    lineText: PropTypes.string,
  }).isRequired,
};

const mapStateToProps = (state) => {
  const {
    isDisplayingAnnotation,
    snippetInformation: {
      lineNumber,
    },
  } = state.annotation;
  // Check that
  const annotation = isDisplayingAnnotation ? (state.app.annotations[lineNumber] && state.app.annotations[lineNumber].annotation) || '' : '';
  return {
    isDisplayingAnnotation,
    readOnly: state.app.readOnly,
    snippetLanguage: state.app.snippetLanguage,
    snippetInformation: state.annotation.snippetInformation,
    annotation,
    appState: state.app,
  };
};

export default connect(mapStateToProps)(Annotations);
