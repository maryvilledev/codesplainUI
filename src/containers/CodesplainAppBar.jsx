import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import {
  AppBar,
  Dialog,
  FlatButton,
} from 'material-ui';
import { withRouter } from 'react-router';
import cookie from 'react-cookie';

import { updateUserSnippets } from '../actions/user';
import { resetState } from '../actions/app';
import { closeAnnotationPanel } from '../actions/annotation';
import LoginButton from '../components/buttons/LoginButton';
import AppMenu from '../components/menus/AppMenu';
import CustomPropTypes from '../util/custom-prop-types';

const CLIENT_ID = process.env.REACT_APP_CLIENT_ID;
const GITHUB_URL = `https://github.com/login/oauth/authorize?client_id=${CLIENT_ID}&scope=read:org`;

const styles = {
  rightElement: {
    marginTop: '16px',
  },
  title: {
    cursor: 'pointer',
  },
};

/*
<CodesplainAppBar /> renders as the banner located at the top of the screen. It
contains the text "Codesplain" on the far left, and either <AppMenu /> or
<LoginButton /> components on its right, depending on whether or not the user
has signed in with GitHub.
*/
export class CodesplainAppBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoggedIn: cookie.load('token') !== undefined,
      isDialogOpen: false,
    };
    this.handleConfirmNavigation = this.handleConfirmNavigation.bind(this);
    this.handleDialogClose = this.handleDialogClose.bind(this);
    this.handleSignOut = this.handleSignOut.bind(this);
    this.handleSnippetSelected = this.handleSnippetSelected.bind(this);
    this.handleTitleTouchTap = this.handleTitleTouchTap.bind(this);
    this.onLoginClick = this.onLoginClick.bind(this);
    this.redirectToHomePage = this.redirectToHomePage.bind(this);
  }

  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(updateUserSnippets());
  }

  onLoginClick() {
    const {
      appState,
      router,
    } = this.props;
    // Saves a cookie containing the current URL path, so we can retrieve it
    // later when GitHub sends the user back to the app, and send the user back
    // to the page they were on when they clicked the login button.
    const body = JSON.stringify(appState);
    cookie.save('signInState', body, { path: '/' });
    cookie.save('signInRedirect', router.location.pathname, { path: '/' });

    // Send the user to the GitHub OAuth authorization URL, where the client ID
    // is set to the CLIENT_ID env var.
    // This does not use the router to route the user to Github because
    // router is used for routing within the application, not the entire web
    window.location = GITHUB_URL;
  }

  handleSignOut() {
    const { router } = this.props;
    cookie.remove('token', { path: '/' });
    cookie.remove('username', { path: '/' });
    cookie.remove('orgs', { path: '/' });
    cookie.remove('userAvatarURL', { path: '/' });
    this.setState({ isLoggedIn: false });
    router.push('/');
    location.reload();
  }

  handleSnippetSelected(key) {
    const username = cookie.load('username');
    window.location = `/${username}/${key}`;
    const { router } = this.props;
    router.push(`/${username}/${key}`);
  }

  handleTitleTouchTap() {
    const { hasUnsavedChanges } = this.props;

    if (hasUnsavedChanges) {
      // Confirm navigation from user
      this.setState({ isDialogOpen: true });
    } else {
      this.redirectToHomePage();
    }
  }

  redirectToHomePage() {
    const {
      dispatch,
      router,
    } = this.props;

    // Reset state
    dispatch(resetState());
    // Close the annotation panel
    dispatch(closeAnnotationPanel());
    // If the user is not already at the home page, redirect them to it
    if (router.location.pathname !== '/') {
      router.push('/');
    }
  }

  handleDialogClose() {
    this.setState({ isDialogOpen: false });
  }

  handleConfirmNavigation() {
    // Close the dialog
    this.handleDialogClose();
    // Redirect the user to the home page
    this.redirectToHomePage();
  }

  render() {
    const actions = [
      <FlatButton
        label="Cancel"
        secondary
        onTouchTap={this.handleDialogClose}
      />,
      <FlatButton
        label="Discard"
        primary
        onTouchTap={this.handleConfirmNavigation}
      />,
    ];

    const { userSnippets } = this.props;
    const { isDialogOpen, isLoggedIn } = this.state;
    const rightElement = isLoggedIn ?
      (<AppMenu
        onSignOut={this.handleSignOut}
        onTitleClicked={this.handleSnippetSelected}
        snippetTitles={userSnippets}
      />)
      : <LoginButton onClick={this.onLoginClick} />;
    const titleElement = (
      <span
        onClick={this.handleTitleTouchTap}
        style={styles.title}
      >
        Codesplain
      </span>
    );

    return (
      <div>
        <AppBar
          showMenuIconButton={false}
          title={titleElement}
          iconElementRight={rightElement}
          iconStyleRight={isLoggedIn ? styles.rightElement : {}}
        />
        <Dialog
          actions={actions}
          modal={false}
          open={isDialogOpen}
          onRequestClose={this.handleDialogClose}
        >
          Discard unsaved changes?
        </Dialog>
      </div>
    );
  }
}

CodesplainAppBar.propTypes = {
  hasUnsavedChanges: PropTypes.bool.isRequired,
  userSnippets: CustomPropTypes.snippets,
};

CodesplainAppBar.defaultProps = {
  userSnippets: {},
};

const mapStateToProps = (state) => {
  const {
    app,
    user: {
      userSnippets,
    },
  } = state;
  return {
    hasUnsavedChanges: app.hasUnsavedChanges,
    appState: app,
    userSnippets,
  };
};

export default withRouter(connect(mapStateToProps)(CodesplainAppBar));
