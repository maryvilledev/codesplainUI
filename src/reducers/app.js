import _ from 'lodash';

import * as actions from '../actions/app';
import { generateFilters } from '../util/codemirror-utils';

export const initialState = {
  annotations: {},
  AST: {},
  filters: {},
  hasUnsavedChanges: false,
  snippetLanguage: 'python3',
  readOnly: false,
  snippet: '',
  snippetKey: '',
  snippetTitle: '',
};

const app = (state = initialState, action) => {
  switch (action.type) {
  case actions.RESET_STATE: {
    return {
      ...initialState,
    };
  }
  case actions.SET_SNIPPET_KEY: {
    return {
      ...state,
      snippetKey: action.payload,
    };
  }
  case actions.SET_SNIPPET_CONTENTS: {
    return {
      ...state,
      hasUnsavedChanges: true,
      snippet: action.payload,
    };
  }
  case actions.SET_RULE_FILTERS: {
    return {
      ...state,
      filters: action.payload,
      hasUnsavedChanges: true,
    };
  }
  case actions.TOGGLE_EDITING_STATE: {
    if (state.readOnly) {
      return state;
    }
    return {
      ...state,
      hasUnsavedChanges: true,
      readOnly: !state.readOnly,
    };
  }
  case actions.SELECT_ALL_FILTERS: {
    const filters = _.mapValues(state.filters, (filter) => {
      if (filter.selected) {
        return filter;
      }
      return { ...filter, selected: true };
    });
    return {
      ...state,
      filters,
    };
  }
  case actions.RESET_RULE_FILTERS: {
    const filters = _.mapValues(state.filters, (filter) => {
      if (filter.selected) {
        return { ...filter, selected: false };
      }
      return filter;
    });
    return {
      ...state,
      filters,
    };
  }
  case actions.SET_AST: {
    return {
      ...state,
      AST: action.payload,
    };
  }
  case actions.SAVE_ANNOTATION: {
    return {
      ...state,
      annotations: {
        ...state.annotations,
        [action.payload.lineNumber]: action.payload,
      },
      hasUnsavedChanges: true,
    };
  }
  case actions.RESTORE_STATE: {
    return {
      ...action.payload,
      hasUnsavedChanges: false,
    };
  }
  case actions.SET_SNIPPET_TITLE: {
    return {
      ...state,
      hasUnsavedChanges: true,
      snippetTitle: action.payload,
    };
  }
  case actions.SAVE_SUCCEEDED: {
    return {
      ...state,
      hasUnsavedChanges: false,
    };
  }
  case actions.SET_SNIPPET_LANGUAGE: {
    return {
      ...state,
      hasUnsavedChanges: true,
      snippetLanguage: action.payload,
    };
  }
  case actions.CLEAR_UNSAVED_CHANGES: {
    return {
      ...state,
      hasUnsavedChanges: false,
    };
  }
  case actions.PARSE_SNIPPET: {
    if (action.meta) {
      return state;
    }
    const {
        AST,
        ruleCounts,
      } = action.payload;
    const filters = generateFilters(state.filters, ruleCounts);
    return {
      ...state,
      AST,
      filters,
    };
  }
  default: {
    return state;
  }
  }
};

export default app;
