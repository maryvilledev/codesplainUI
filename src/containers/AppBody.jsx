import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import cookie from 'react-cookie';

import Annotations from './Annotations';
import FilterArea from './FilterArea';
import SnippetArea from './SnippetArea';
import {
  loadSnippet,
  restoreState,
} from '../actions/app';
import {
  switchOrg,
} from '../actions/user';
import { setPermissions } from '../actions/permissions';
import NotFound from '../components/NotFound';
import { removeDeprecatedFiltersFromState } from '../util/codemirror-utils';
import { sanitizeKey } from '../util/requests';
import { setDefaults } from '../util/state-management';

const styles = {
  body: {
    display: 'flex',
    flexFlow: 'row wrap',
    height: '100%',
  },
  rightSection: {
    display: 'flex',
    flexFlow: 'column nowrap',
    height: '100%',
    flex: '2 1',
    overflowY: 'auto',
    boxShadow: '0px 1px 6px rgba(0, 0, 0, 0.12), 0px 1px 4px rgba(0, 0, 0, 0.12)',
  },
};

export class AppBody extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isValidSnippet: true,
    };
    this.loadSnippet = this.loadSnippet.bind(this);
    this.updatePermissions = this.updatePermissions.bind(this);
  }

  componentDidMount() {
    this.loadSnippet();
  }

  componentWillReceiveProps(nextProps) {
    // Reload snippet if URL changes. Assume it is valid, and
    // if it is not, the call to loadSnippet will handle it.
    if (this.state.pathname !== nextProps.router.location.pathname) {
      this.setState({ isValidSnippet: true });
      this.loadSnippet();
    }
  }

  componentDidUpdate() {
    const { router } = this.props;
    const {
      id: snippetKey,
      username: snippetOwner,
    } = router.params;

    if (snippetOwner && snippetKey) {
      this.updatePermissions();
    }
  }

  loadSnippet() {
    const {
      dispatch,
      router,
    } = this.props;
    const {
      id: snippetKey,
      username: snippetOwner,
    } = router.params;

    this.setState({ pathname: router.location.pathname });
    if (!snippetOwner && !snippetKey) {
      // This is a new snippet for the current user, enable all permissions
      const permissions = {
        canRead: true,
        canEdit: true,
      };
      dispatch(setPermissions(permissions));

      // Reload the state if returning from login.
      const signInState = cookie.load('signInState');
      if (signInState) {
        cookie.remove('signInState', { path: '/' });
        cookie.remove('signInRedirect', { path: '/' });
        dispatch(restoreState(signInState));
      }
      return;
    }

    dispatch(loadSnippet(snippetOwner, snippetKey))
      .then((res) => {
        // Normalize the app state received from S3
        const appState = setDefaults(removeDeprecatedFiltersFromState(res.data));
        // Snippet may not have a S3 key value saved; set it to the URL's id
        // param if the state object is lacking it
        if (!appState.snippetKey) {
          appState.snippetKey = snippetKey;
        }
        // Restore the application's state
        dispatch(restoreState(appState));
        this.updatePermissions();

        // Reroute if using legacy url
        // So /:username/snippets/:id -> /:username/:id
        const nextRoute = `/${snippetOwner}/${sanitizeKey(snippetKey)}`;
        if (router.location.pathname !== nextRoute) {
          router.push(nextRoute);
        }
      }, () => {
        // Failed to get the snippet, either bad URL or unauthorized
        this.setState({
          isValidSnippet: false,
        });
      });
  }

  updatePermissions() {
    const {
      username,
      orgs,
      dispatch,
      router,
    } = this.props;
    const {
      username: snippetOwner,
    } = router.params;

    // If the user is a member of the org in question, make the org the
    // current org
    const isMember = orgs.includes(snippetOwner);
    if (isMember) {
      dispatch(switchOrg(snippetOwner));
    }

    const permissions = {
      canRead: true,
      // Users may only edit & save a file they (or one of their orgs) own
      canEdit: (username === snippetOwner || isMember),
    };
    dispatch(setPermissions(permissions));
  }

  render() {
    const { isValidSnippet } = this.state;
    if (!isValidSnippet) {
      return <NotFound />;
    }
    return (
      <div style={styles.body}>
        <SnippetArea />
        <div style={styles.rightSection}>
          <FilterArea />
          <Annotations />
        </div>
      </div>
    );
  }
}

AppBody.propTypes = {
  orgs: PropTypes.string,
  username: PropTypes.string,
};

AppBody.defaultProps = {
  orgs: [],
  username: '',
};

const mapStateToProps = (state) => {
  const {
    user: {
      username,
      orgs,
    },
  } = state;

  return {
    username,
    orgs,
  };
};

export default withRouter(connect(mapStateToProps)(AppBody));
