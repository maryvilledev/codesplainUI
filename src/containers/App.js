import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import injectTapEventPlugin from 'react-tap-event-plugin';

import AppBody from './AppBody';
import CodesplainAppBar from './CodesplainAppBar';
import { Router, Route, browserHistory } from 'react-router'

// Needed for onTouchTap
// http://stackoverflow.com/a/34015469/988941
injectTapEventPlugin();

class App extends Component {
  render() {
    const component = (props) => (
      <MuiThemeProvider>
        <div
          className="container-fluid"
        >
          <CodesplainAppBar />
          <AppBody {...props}/>
        </div>
      </MuiThemeProvider>
    )
    return (
      <Router history={browserHistory}>
        <Route path="/(:id)" component={component} />
      </Router>
    );
  }
}

export default App;
