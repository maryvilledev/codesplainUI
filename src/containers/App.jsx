import React from 'react';

import AppBody from './AppBody';
import CodesplainAppBar from './CodesplainAppBar';
import NotificationSnackbar from '../components/NotificationSnackbar';

export const App = () => (
  <div className="container-fluid">
    <CodesplainAppBar />
    <AppBody />
    <NotificationSnackbar />
  </div>
);

export default App;
