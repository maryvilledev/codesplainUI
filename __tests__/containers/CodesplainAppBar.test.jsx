import React from 'react';
import cookie from 'react-cookie';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import { createStore } from 'redux';
import app from '../../src/reducers/app.js'

import { CodesplainAppBar } from '../../src/containers/CodesplainAppBar';

describe('<CodesplainAppBar />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  describe('snapshot tests', () => {
    afterEach(() => {
      cookie.remove('token');
    });
    it('user is not logged in', () => {
      const wrapper = shallowWithContext(
        <CodesplainAppBar
          store={createStore(app)}
        />
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('user is logged in', () => {
      cookie.save('token', 'foobar')
      const wrapper = shallowWithContext(
        <CodesplainAppBar
          store={createStore(app)}
        />
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
