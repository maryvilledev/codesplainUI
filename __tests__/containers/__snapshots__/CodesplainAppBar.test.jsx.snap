exports[`<CodesplainAppBar /> snapshot tests user is logged in 1`] = `
<div>
  <AppBar
    iconElementRight={
      <AppMenu
        onSignOut={[Function]}
        onTitleClicked={[Function]}
        snippetTitles={Object {}} />
    }
    showMenuIconButton={false}
    title={
      <span
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }>
        Codesplain
      </span>
    }
    zDepth={1} />
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={false}
          secondary={true} />,
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Discard"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false} />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}>
    Discard unsaved changes?
  </Dialog>
</div>
`;

exports[`<CodesplainAppBar /> snapshot tests user is not logged in 1`] = `
<div>
  <AppBar
    iconElementRight={
      <LoginButton
        onClick={[Function]} />
    }
    showMenuIconButton={false}
    title={
      <span
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }>
        Codesplain
      </span>
    }
    zDepth={1} />
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={false}
          secondary={true} />,
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="Discard"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false} />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}>
    Discard unsaved changes?
  </Dialog>
</div>
`;
